src/main/java/ui/components/FilterTextField.java

```diff
@@ -55,12 +55,19 @@ private void setup() {
         validationSupport.registerValidator(this, (c, newValue) -> {
             boolean wasError = false;
             try {
-                Parser.parse(getText());
-                setStyleForValidFilter();
+                if (!Parser.check(getText())) {
+                    wasError = true;
+                }
```

Comment:
Naming the variable as `isError` should be better? Also, what is the advantage of having this `check` method instead of catching 2 types of exceptions here?